File: src/2022/day2/p2.js
const readFileSync = require('fs').readFileSync;

const data = readFileSync('./input.txt', 'utf8')

const lines = data.split(/\n/);

const myMoves = {
    //rock
    "X": 1,
    //paper
    "Y": 2,
    //scissors
    "Z": 3
}

const resultScore = {
    "X": 0,
    "Y": 3,
    "Z": 6
}

const matchScore = {
    //rock
    "A": {
        //lose
        "X": "Z",
        //draw
        "Y": "X",
        //win
        "Z": "Y"
    },
    //paper
    "B": {
        "X": "X",
        "Y": "Y",
        "Z": "Z"
    },
    //scissors
    "C": {
        "X": "Y",
        "Y": "Z",
        "Z": "X"
    }
}

const score = lines.reduce((currScore, currLine) => {
    return (
        currScore + resultScore[currLine[2]] + myMoves[matchScore[currLine[0]][currLine[2]]]
    )
}, 0)

console.log(score);

File: src/2022/day2/p2.go
package main

import ("fmt";"bufio";"log";"os")

func main() {
	score := 0;
	myMoveScore := map[string]int {
		"X": 1,	//rock
		"Y": 2,	//paper
		"Z": 3,	//scissors
	}
	loseWinDrawScore := map[string]int {
		"X": 0,
		"Y": 3,
		"Z": 6,
	}
	matchOutcomeScore := map[string]map[string]string {
		"A": {	//rock
			"X": "Z",	//lose
			"Y": "X",	//draw
			"Z": "Y",	//win
		},
		"B": {	//paper
			"X": "X",
			"Y": "Y",
			"Z": "Z",
		},
		"C": {	//scissors
			"X": "Y",
			"Y": "Z",
			"Z": "X",
		},
	}
	file, err1 := os.Open("./input.txt")
	if err1 != nil {log.Fatal(err1)}
	defer file.Close()
	scanner := bufio.NewScanner(file)

	for scanner.Scan() {
		currEnemyMove := string(scanner.Text()[0])
		currMyMove := string(scanner.Text()[2])
		score += loseWinDrawScore[currMyMove]
		score += myMoveScore[matchOutcomeScore[currEnemyMove][currMyMove]]
	}

	fmt.Println(score)
}